ODIN_II/SRC/netlist_check.cpp

```diff
@@ -717,3 +721,90 @@ nnode_t* find_node_at_top_of_combo_loop(nnode_t* start_node) {
         }
     }
 }
+
+/**
+ * @brief This function travseres the netlist both forwards and backwards and keeps the record of nodes with same type in each traverse.
+ * In case that there is a mismatch between the number of nodes counted in both traverses, It will throw a warning to
+ * indicate a discontinuity in the netlist.
+ */
+void check_discontinuity(netlist_t* netlist) {
+    //Declare variables that hold number of elements
+    long long num_of_forward_elements{0}, num_of_backward_elements{0};
+    //Declare type of the node
+    operation_list op{MULTIPLY};
+
+    //Forward traverse starting from top input nodes
+    for (int i = 0; i < netlist->num_top_input_nodes; i++) {
+        discontinuity_check_forward_traverse(netlist->top_input_nodes[i], op, num_of_forward_elements, NETLIST_DISCONTINUITY_TRAVERSE_VALUE);
+    }
+
+    //Backward traverse statring from top output nodes
+    for (int i = 0; i < netlist->num_top_output_nodes; i++) {
+        discontinuity_check_backward_traverse(netlist->top_output_nodes[i], op, num_of_backward_elements, NETLIST_DISCONTINUITY_TRAVERSE_VALUE + 1);
+    }
+
+    //Detect discontinuity
+    if (num_of_forward_elements != num_of_backward_elements) {
+        printf("\n\x1B[33mWARNING! discontinuity detected within the netlist.\n");
```

Comment:
There are `error_message`, and `warning_message` routines in the Odin-II codebase. Please have a look at them and use them appropriately when required.

Reply:
Applied as suggested.